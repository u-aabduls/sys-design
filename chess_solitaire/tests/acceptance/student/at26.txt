-- ACCEPTANCE TEST #26: VIOLATION CASE: moves - slot_2_occupied

-- Setup some chess pieces for the game
setup_chess (K,1,3)
setup_chess (Q,2,4)

-- attempt to move and capture before game has started
move_and_capture(1,-2,2,3)

-- start game
start_game

-- attempt to move and capture from an invalid slot_1
move_and_capture(1,-2,2,3)

-- attempt to move and capture from an invalid slot_2
move_and_capture(1,2,2,-3)

-- attempt to move and capture from an unoccupied slot_1
move_and_capture(1,2,2,3)

-- attempt to move and capture to an unoccupied slot_2
move_and_capture(1,3,2,3)