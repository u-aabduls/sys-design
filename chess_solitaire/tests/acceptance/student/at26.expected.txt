  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(K,1,3)
  # of chess pieces on board: 1
  Game being Setup...
  ..K.
  ....
  ....
  ....
->setup_chess(Q,2,4)
  # of chess pieces on board: 2
  Game being Setup...
  ..K.
  ...Q
  ....
  ....
->move_and_capture(1,-2,2,3)
  # of chess pieces on board: 2
  Error: Game not yet started
  ..K.
  ...Q
  ....
  ....
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ..K.
  ...Q
  ....
  ....
->move_and_capture(1,-2,2,3)
  # of chess pieces on board: 2
  Error: (1, -2) not a valid slot
  ..K.
  ...Q
  ....
  ....
->move_and_capture(1,2,2,-3)
  # of chess pieces on board: 2
  Error: (2, -3) not a valid slot
  ..K.
  ...Q
  ....
  ....
->move_and_capture(1,2,2,3)
  # of chess pieces on board: 2
  Error: Slot @ (1, 2) not occupied
  ..K.
  ...Q
  ....
  ....
->move_and_capture(1,3,2,3)
  # of chess pieces on board: 2
  Error: Slot @ (2, 3) not occupied
  ..K.
  ...Q
  ....
  ....
